### PR TITLE
Add Friends Functionality

### PR DESCRIPTION
**Pull request by redls**
Added Friends APIs: Play Games now has a cross-game Friends list. Request access to your players cross-game PGS friends list and integrate it with your in-game system to help players play with a set of high-quality gaming friends, and help people find their friends within your game. Learn more about Play Games Services .

### CODE REVIEW

**Review by olehkuznetsov — COMMENTED** [review]
Why do we remove android here? and previously for background?

File: ButtonClicker/build.gradle [hunk 1]
```diff
@@ -22,18 +22,22 @@ android {
...
     }
 }
 
 dependencies {
-    implementation "com.android.support:appcompat-v7:${appcompat_library_version}"
-    implementation "com.android.support:support-v4:${support_library_version}"
-    implementation "com.google.android.gms:play-services-games:${gms_library_version}"
+    implementation 'androidx.appcompat:appcompat:1.0.0'
+    implementation 'androidx.legacy:legacy-support-v4:1.0.0'
+    implementation "com.google.android.gms:play-services-games:20.0.0"
```
Comment (olehkuznetsov) [inline ButtonClicker/build.gradle:35]: 20.0.1?

File: ButtonClicker/src/main/res/values/styles.xml [hunk 2]
```diff
@@ -48,7 +48,7 @@
 
 
     <style name="InvPopup" parent="@style/LLScreen">
-        <item name="android:background">@color/PopupColor</item>
+        <item name="background">@color/PopupColor</item>
```
Comment (olehkuznetsov) [inline ButtonClicker/src/main/res/values/styles.xml:51]: why it become just background and other left with android?

File: TypeANumber/src/main/java/com/google/example/games/tanc/FriendsFragment.java [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+import java.util.ArrayList;
+
+/** Fragment that shows the list of friends. */
+public class FriendsFragment extends Fragment {
+    private static final String TAG = "FriendsFragment";
+    private static final int PAGE_SIZE = 200;
+
+    interface Listener {
+        // called when the user presses the `Back` button
+        void onBackButtonClicked();
```
Comment (olehkuznetsov) [inline TypeANumber/src/main/java/com/google/example/games/tanc/FriendsFragment.java:41]: wrong code format

**Review by olehkuznetsov — CHANGES_REQUESTED** [review]

File: TypeANumber/src/main/java/com/google/example/games/tanc/MainActivity.java [hunk 4]
```diff
@@ -99,27 +116,33 @@ public void onCreate(Bundle savedInstanceState) {
     setContentView(R.layout.activity_main);
 
     // Create the client used to sign in to Google services.
-    mGoogleSignInClient = GoogleSignIn.getClient(this,
-        new GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_GAMES_SIGN_IN).build());
+    mGoogleSignInClient =
```
Comment (olehkuznetsov) [inline TypeANumber/src/main/java/com/google/example/games/tanc/MainActivity.java:119]: Do we need this change?

Comment (redls) [reply TypeANumber/src/main/java/com/google/example/games/tanc/MainActivity.java]: fixed

**Pull request merged**